### DISCUSSION TITLE
"Draw.MolsToGridImage" doesn't show molecule drawings when I use it in a function

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4060] [Q&A] [answered]
Hi, thank you very much for your time to read my question. 

For programming purpose, I need to define a function to visualize smiles. I am using the `Draw.MolsToGridImage` to draw smiles as below:

```
smiles = ['OCCc1ccn2cnccc12','C1CC1Oc1cc2ccncn2c1']
mols = [Chem.MolFromSmiles(smile) for smile in smiles]
Draw.MolsToGridImage(mols, molsPerRow=2, subImgSize=(200, 200), legends=smiles)
```

When I define a function:

```
smiles = ['OCCc1ccn2cnccc12','C1CC1Oc1cc2ccncn2c1']

def visualize(smiles):
    mols = [Chem.MolFromSmiles(smile) for smile in smiles]
    Draw.MolsToGridImage(mols, molsPerRow=2, subImgSize=(200, 200), legends=smiles)
```
 
Calling function `visualize(smiles)` in Jupyter Notebook doesn't show any drawings of molecules. 

I can save output of `Draw.MolsToGridImage` into a variable named "image" and return it from function. In this case, function `visualize(smiles)` doesn't show molecule images too. I need to type "image" in  a new cell to see the molecule images. 
 
```
smiles = ['OCCc1ccn2cnccc12','C1CC1Oc1cc2ccncn2c1']

def visualize(smiles):
    mols = [Chem.MolFromSmiles(smile) for smile in smiles]
    image = Draw.MolsToGridImage(mols, molsPerRow=2, subImgSize=(200, 200), legends=smiles)
    return image

image = visualize(smiles)
```

If you have any idea how to solve this issue, I really appreciate it if you help me. Thank you.

**Comment by kexul** [comment]
If you're using jupyter notebook, you may try add the following line: 
```
%matplotlib inline 
```

**Comment by apahl** [comment, marked as answer]
If you just call `visualize(smiles)` in your second example (without assigning it to a variable), it should display the molecules.

This has the same effect as 
```
image = visualize(smiles)

# New cell
image
```
which you say, does work.

KR, Axel

**Reply by mohammad-saber** [reply]
Thank you.  It works.
